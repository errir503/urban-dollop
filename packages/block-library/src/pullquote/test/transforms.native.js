/**
 * External dependencies
 */
import {
	getEditorHtml,
	initializeEditor,
	setupCoreBlocks,
	transformBlock,
	getBlockTransformOptions,
} from 'test/helpers';

const block = 'Pullquote';
const initialHtml = `
<!-- wp:pullquote -->
<figure class="wp-block-pullquote"><blockquote><p>One of the hardest things to do in technology is disrupt yourself.</p><cite>Matt Mullenweg</cite></blockquote></figure>
<!-- /wp:pullquote -->`;

const transformsWithInnerBlocks = [ 'Quote', 'Columns', 'Group' ];
const blockTransforms = [
	'Paragraph',
	'Heading',
	...transformsWithInnerBlocks,
];

setupCoreBlocks();

describe( `${ block } block transforms`, () => {
	test.each( blockTransforms )( 'to %s block', async ( blockTransform ) => {
		const screen = await initializeEditor( { initialHtml } );
		const newBlock = await transformBlock( screen, block, blockTransform, {
			hasInnerBlocks:
				transformsWithInnerBlocks.includes( blockTransform ),
		} );
		expect( newBlock ).toBeVisible();
		expect( getEditorHtml() ).toMatchSnapshot();
	} );

	it( 'matches expected transformation options', async () => {
		const screen = await initializeEditor( { initialHtml } );
		const transformOptions = await getBlockTransformOptions(
			screen,
			block
		);
		expect( transformOptions ).toHaveLength( blockTransforms.length );
	} );
} );
