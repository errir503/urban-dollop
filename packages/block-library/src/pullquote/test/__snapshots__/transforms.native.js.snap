// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pullquote block transforms to Columns block 1`] = `
"<!-- wp:columns -->
<div class="wp-block-columns"><!-- wp:column {"width":"100%"} -->
<div class="wp-block-column" style="flex-basis:100%"><!-- wp:pullquote -->
<figure class="wp-block-pullquote"><blockquote><p>One of the hardest things to do in technology is disrupt yourself.</p><cite>Matt Mullenweg</cite></blockquote></figure>
<!-- /wp:pullquote --></div>
<!-- /wp:column --></div>
<!-- /wp:columns -->"
`;

exports[`Pullquote block transforms to Group block 1`] = `
"<!-- wp:group {"layout":{"type":"constrained"}} -->
<div class="wp-block-group"><!-- wp:pullquote -->
<figure class="wp-block-pullquote"><blockquote><p>One of the hardest things to do in technology is disrupt yourself.</p><cite>Matt Mullenweg</cite></blockquote></figure>
<!-- /wp:pullquote --></div>
<!-- /wp:group -->"
`;

exports[`Pullquote block transforms to Heading block 1`] = `
"<!-- wp:heading -->
<h2 class="wp-block-heading">One of the hardest things to do in technology is disrupt yourself.</h2>
<!-- /wp:heading -->

<!-- wp:heading -->
<h2 class="wp-block-heading">Matt Mullenweg</h2>
<!-- /wp:heading -->"
`;

exports[`Pullquote block transforms to Paragraph block 1`] = `
"<!-- wp:paragraph -->
<p>One of the hardest things to do in technology is disrupt yourself.</p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p>Matt Mullenweg</p>
<!-- /wp:paragraph -->"
`;

exports[`Pullquote block transforms to Quote block 1`] = `
"<!-- wp:quote -->
<blockquote class="wp-block-quote"><!-- wp:paragraph -->
<p>One of the hardest things to do in technology is disrupt yourself.</p>
<!-- /wp:paragraph --><cite>Matt Mullenweg</cite></blockquote>
<!-- /wp:quote -->"
`;
